make/modules/java.base/Lib.gmk

```diff
@@ -242,8 +242,8 @@ ifeq ($(call isTargetOs, linux)+$(call isTargetCpu, x86_64)+$(INCLUDE_COMPILER2)
       NAME := simdsort, \
       TOOLCHAIN := TOOLCHAIN_LINK_CXX, \
       OPTIMIZATION := HIGH, \
-      CFLAGS := $(CFLAGS_JDKLIB), \
-      CXXFLAGS := $(CXXFLAGS_JDKLIB), \
+      CFLAGS := $(CFLAGS_JDKLIB) -std=c++17, \
```

Comment:
Hi Magnus, the new x86-simd-sort 4.0 needs C++17 to compile. Will look into the changes needed for this library to compile without the C++17 standard and get back to you.

Thanks,
Vamsi

Reply:
Seems a bit odd to me too as the existing simd code seems to C code residing in .cpp files for some reason.

Reply:
Yeah, that is also a good point. Why setting C++ flags in CFLAGS, and not just CXXFLAGS?

Also @vamsi-parasa, did you check into if this really was needed and why? To be clear, I basically consider this an integration blocker.

Reply:
Hi Magnus (@magicus),
The `-std=c++17` flag was to be placed in CXXFLAGS. Will fix that soon. 
Also, the latest version of x86-simd-sort is using C++17 features (which are supported from GCC 8 onwards). Would it be possible to restrict the availability of simdsort library in OpenJDK to GCC version >= 8 ?

Reply:
Raising the minimum gcc version is not done willy-nilly. (I feel a "You just don't ..." meme coming up)

But you are saying that you want to skip building this library unless you have a gcc version that supports c++17? 

I still don't really like it. I'd like to hear someone else who can think clearly about this, if we want to go down this path, and start adding libraries that use C++17. Maybe @kimbarrett has some input?

Reply:
Then you must add logic to check for this. Now the build will just fail if building with an older gcc. That is not acceptable.

Reply:
Hi Marcus (@magicus), please see the updated code which added guards to check for GCC version >= 7.5 in src/java.base/linux/native/libsimdsort/{avx2-linux-qsort.cpp, avx512-linux-qsort.cpp}. Made sure that OpenJDK builds without errors using both GCC 7.5 and GCC 6.4.

Reply:
That sounds weird. You can't check for if compiler options should be enabled or not inside source code files.

Are you saying that when compiling with GCC 6, it will just silently ignore `-std=c++17`? I'd have assumed that it printed a warning or error about an unknown or invalid option, if C++17 is not supported.

Reply:
Hi Magnus (@magicus),
 
> Are you saying that when compiling with GCC 6, it will just silently ignore `-std=c++17`? I'd have assumed that it printed a warning or error about an unknown or invalid option, if C++17 is not supported.

The GCC complier for versions 6 (and even 5) silently ignores the flag `-std=c++17`. It does not print any warning or error. I tested it with a toy C++ program and also by building OpenJDK using GCC 6. 

> You can't check for if compiler options should be enabled or not inside source code files.

 what I meant was, there are #ifdef guards using predefined macros in the C++ source code to check for GCC version and make the simdsort code available for compilation or not based on the GCC version

```cpp
// src/java.base/linux/native/libsimdsort/simdsort-support.hpp
#if defined(_LP64) && (defined(__GNUC__) && ((__GNUC__ > 7) || ((__GNUC__ == 7) && (__GNUC_MINOR__ >= 5))))
        #define __SIMDSORT_SUPPORTED_LINUX
#endif
```

```cpp
//src/java.base/linux/native/libsimdsort/avx2-linux-qsort.cpp
#include "simdsort-support.hpp"
#ifdef __SIMDSORT_SUPPORTED_LINUX
            <simdsort functions>
#endif
```

Reply:
Okay, then I guess I am fine with this.

Reply:
Thank you Magnus!

Reply:
@vamsi-parasa You said:
> Made sure that OpenJDK builds without errors using both GCC 7.5 and GCC 6.4.

but now we have https://bugs.openjdk.org/browse/JDK-8321688. Did you introduce any changes after you tested with GCC 7.5? It seems strange to me that the code simultaneously both works and not works with gcc 7.5.

Reply:
Hi Magnus (@magicus), did a fresh pull of the OpenJDK and was able to build it successfully (without any errors) using GCC 7.5.0 on Ubuntu Linux machine. 
(I am on vacation till Jan7th, 2024. Our team will look into this issue)

Reply:
New information in JDK-8321688 says it is only happening on slowdebug. This is probably why it was missed.